### ISSUE TITLE
Enhancement: Integrate Authentication with Swagger Docs

### ISSUE THREAD

**Issue by cofin** [starlite-api/starlite#397]
Enable OpenAPI generated Swagger docs should be aware of API authentication mechanism to allow for easy testing from the web browser.

Here is a sample from a FastAPI project that shows the desired functionality:
<img width="751" alt="Screen Shot 2022-08-22 at 9 16 00 PM" src="https://user-images.githubusercontent.com/204685/186054390-efc6f888-f3a3-4375-be0d-15e31a9b1f9a.png">
<img width="677" alt="Screen Shot 2022-08-22 at 9 15 30 PM" src="https://user-images.githubusercontent.com/204685/186054468-7dee5009-f905-4fbd-be95-410a445e522b.png">

After authentication, the user should be able to invoke services with the credentials. 

API Key & OAUTH2 support are included out of the box.  Could be extended to support OpenID Connect

**Comment by toudi** [comment]
I stumbled upon this issue because I tried to generate openapi specs and realized that something wasn't right. Anyhow, I was able to move just one step further with this piece of code:

```python
from pydantic_openapi_schema.v3_1_0.security_scheme import SecurityScheme

TokenSecurity: SecurityScheme = SecurityScheme(
    type="apiKey", description="Token based authentication", schema="bearer"
)

class CustomOpenAPIConfig(OpenAPIConfig):
    def to_openapi_schema(self) -> OpenAPI:
        schema = super().to_openapi_schema()
        schema.components = Components(securitySchemes={"BearerAuth": TokenSecurity})

        return schema

app = Starlite(
    # ....
    openapi_config=CustomOpenAPIConfig( .... ),
)

```

however I still do not understand how am I supposed to decorate particular views with `security` attribute. the openAPI docs specify this:

(via https://swagger.io/docs/specification/authentication/ )

```yaml
paths:
  /billing_info:
    get:
      summary: Gets the account billing info
      security:
        - OAuth2: [admin]   # Use OAuth with a different scope
```
however I looked in Router, the `get` decorator and I couldn't find this parameter. Is it because this is simply not implemented yet ?

**Comment by Goldziher** [comment]
> I stumbled upon this issue because I tried to generate openapi specs and realized that something wasn't right. Anyhow, I was able to move just one step further with this piece of code:
> 
> ```python
> from pydantic_openapi_schema.v3_1_0.security_scheme import SecurityScheme
> 
> TokenSecurity: SecurityScheme = SecurityScheme(
>     type="apiKey", description="Token based authentication", schema="bearer"
> )
> 
> class CustomOpenAPIConfig(OpenAPIConfig):
>     def to_openapi_schema(self) -> OpenAPI:
>         schema = super().to_openapi_schema()
>         schema.components = Components(securitySchemes={"BearerAuth": TokenSecurity})
> 
>         return schema
> 
> app = Starlite(
>     # ....
>     openapi_config=CustomOpenAPIConfig( .... ),
> )
> ```
> 
> however I still do not understand how am I supposed to decorate particular views with `security` attribute. the openAPI docs specify this:
> 
> (via https://swagger.io/docs/specification/authentication/ )
> 
> ```yaml
> paths:
>   /billing_info:
>     get:
>       summary: Gets the account billing info
>       security:
>         - OAuth2: [admin]   # Use OAuth with a different scope
> ```
> 
> however I looked in Router, the `get` decorator and I couldn't find this parameter. Is it because this is simply not implemented yet ?

Yup, this is not implemented. It should be - so if you feel like adding a PR it would be welcome, otherwise it will be added in the nearish future.

**Comment by toudi** [comment]
wow I got it to work :D I'll submit a PR, at least it will be something to expand on :)

**Comment by cofin** [comment]
closed by #400
